app/javascript/react/utils/urlShortenedLink.ts

```diff
@@ -0,0 +1,33 @@
+import axios from "axios";
+import { useEffect, useState } from "react";
+
+const useShortenedLink = (url: string, accessToken: string) => {
+  const [shortenedLink, setShortenedLink] = useState<string>(url);
+  const [error, setError] = useState<Error | null>(null);
+
+  useEffect(() => {
+    const shortenLink = async () => {
+      try {
+        const response = await axios.post(
```

Comment:
It could be helpful to add an interface with Bitly Response to define the expected structure of the response, so we can catch issues early